modules/core/src/main/scala/org/scalasteward/core/buildtool/mill/parser.scala

```diff
@@ -43,6 +47,62 @@ object parser {
   def parseMillVersion(s: String): Option[Version] =
     Option(s.trim).filter(_.nonEmpty).map(Version.apply)
 
+  /**
+    * Used to correctly format the Mill plugin artifacts will when included look like:
+    * - import $ivy.`com.goyeau::mill-scalafix::0.2.10`
+    * However for the actual artifact as of 0.10.x this will look like:
+    * - mill-scalafix_mill0.10_.2.13
+    *
+    * @param artifactName name of the artifact parsed from the build file
+    * @param millVerion the current Mill version being used
+    * @return the newly put together ArtifactId
+    */
+  private def millPluginArtifact(artifactName: String, millVersion: Version): ArtifactId = {
+    val millSuffix = millVersion.value.split('.') match {
+      // Basically for right now we only accept "0.10.x" which is when this syntax for Mill plugins
+      // was introduced. As the versions increase we'll need to readdress this.
```

Comment:
This was backported to Mill 0.9.12 and will be used most likely in future 0.x versions of Mill.
Why don't you accept any major and minor instead?

Reply:
Mainly because if I accept any version, there is no guarantee that it will work. I've seen a lot of 0.9.x repos that were using the old format before the whole `_mill*` thing was introduced, so this wouldn't work on those. Same with 1.x stuff, there is no guarantee that this will be the same patter as mentioned [here](https://github.com/com-lihaoyi/mill/pull/1994#discussion_r951594263). So I guess I was just being a bit conservative here. I can switch it, but then it should probably only match on `0.9.12` if it's a `0.9.x` and only consider minors until we're positive this will be the format for `1.x`.

wdyt?

Reply:
I went ahead and did that, I included 0.9.12, expanded it to any minor above that, but still don't consider 1.x until we are 100% sure it'll be this way.

Reply:
You should only check the dependencies that have the second `::`.
These are accepted only in versions of Mill that support the binary suffix so it is safe in older versions since mill plugins were imported with:
```scala
import $ivy.`group_id::artifact_id:version`
```
And after the mill binary syntax was introduced you can have:
```scala
import $ivy.`group_id::artifact_id::version`
```

Note also that even in current mill, some plugins didn't update yet to the new `_mill0.10` binary suffix and need to be imported without the second `::`

Reply:
About `1.x` we can't see the future, but most likely this will not change.
I don't see other ways `group_id::artifact::version` could be used if not for mill binary version.
If you make it general, it will work out of the box in case Mill doesn't change.
If you fix it to `0.x` you have 100% probability of future maintenance to implement it for 1.x.

Reply:
> You should only check the dependencies that have the second ::.

Yes, this was intended actually but I had leftover stuff in the parser things. I've updated this now to look for the `::` instead of just `:` so it will only parse those that use the double `::`.

Reply:
Actually re-reading this, sorry I don't understand what you mean by:

> Note also that even in current mill, some plugins didn't update yet to the new _mill0.10 binary suffix and need to be imported without the second ::

If they haven't updated to this format, then it'd be `:`, not `::` right?

Reply:
> Actually re-reading this, sorry I don't understand what you mean by:
> 
> > Note also that even in current mill, some plugins didn't update yet to the new _mill0.10 binary suffix and need to be imported without the second ::
> 
> If they haven't updated to this format, then it'd be `:`, not `::` right?

Right!

```scala
import $ivy.`org.example::plugin:1.2.3`
```
fetches `org.example:plugin_2.13:1.2.3`

```scala
import $ivy.`org.example::plugin::1.2.3`
```
fetches `org.example:plugin_mill0.10_2.13:1.2.3`